clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -488,7 +491,11 @@ private void close(KafkaChannel channel) {
         } catch (IOException e) {
             log.error("Exception closing connection to node {}:", channel.id(), e);
         }
-        this.stagedReceives.remove(channel);
+        Deque<NetworkReceive> deque = this.stagedReceives.remove(channel);
+        if (deque != null) {
+            addToCompletedReceives(channel, deque, true);
```

Comment:
I suppose, but since brokers don't attempt to complete client requests during controlled shutdown, responses may or may not be sent anyway. So I didn't think it was worth handling this case.